core/server/worker/src/main/java/alluxio/worker/block/BlockMasterSync.java

```diff
@@ -129,7 +125,8 @@ private void registerWithMaster() throws IOException {
     List<ConfigProperty> configList =
         Configuration.getConfiguration(Scope.WORKER);
 
-    if (ACQUIRE_LEASE) {
+    boolean leaseRequired = Configuration.getBoolean(PropertyKey.WORKER_REGISTER_LEASE_ENABLED);
+    if (leaseRequired) {
```

Comment:
This 'leaseRequired' field is inlined to unblock testing different configurations. Also I believe this is the expected behavior of configurations: after global configuration changed, the behavior of the classes should not adhere to a stale snapshot captured in a static final variable.